js-pkg/sdk/src/main.ts

```diff
@@ -135,6 +135,18 @@ export class DocumentManager {
     const clientToken = await this.getClientToken(docId, authDocRequest)
     return new DocConnection(clientToken)
   }
+
+  /**
+   * Creates a new document with initial content.
+   *
+   * @param update A Yjs update byte string representing the initial content.
+   * @returns A {@link DocCreationResult} object containing the ID of the created document.
+   */
+  public async createDocWithContent(update: Uint8Array): Promise<DocCreationResult> {
```

Comment:
We allow users to create documents with a given ID, but we aren't taking a `docId` param here. I expect we'll have users who want to do this since [the cleanest way to implement the auth endpoint](getOrCreateDocAndToken(docId)) is for the user to provide a doc ID. At the same time, calling this with a `docId` that already exists would result in some unexpected behavior (adding an update to a doc that already has content). I feel like the correct implementation of a `createDocWithContent` function would error if the doc already exists, but I don't think we can do that from the client (and it may not be easy to do it from the server either). So I guess we have two options here:

1. leave as is and don't allow users to provide a `docId` possibly forcing users to keep their own mapping to a doc ID
2. allow users to provide a `docId` but risk them accidentally (and without warning) pushing updates to an existing doc

@michaelsilver @paulgb @feliciachang thoughts?

Reply:
I made this explicit design choice based on @paulgb’s input but we can change course if we want

Reply:
Yes, the thinking here was that for now the only way we can guarantee that we are creating a new document is if we generate the doc ID from Y-Sweet. If we allow users to set their own ID, and they set it to a document that already exists, the initial document content will be added into the existing document, which may cause unexpected results (for example, if the initial document is a template for a text doc, it will add the template again to the existing doc)

To alleviate this down the road, I think we should have a server-side endpoint that applies the initial content _only_ if the document is actually new. But we don't have that yet, so we resort to this.